common/base/src/base/string_func.rs

```diff
@@ -103,3 +103,34 @@ pub fn mask_string(s: &str, unmask_len: usize) -> String {
         ret
     }
 }
+
+/// Replace idx-th char as new char
+/// If idx is out of len(s) range, then no replacement is performed.
+/// replace_nth_char("a13", 1, '2') -> 'a23'
+/// replace_nth_char("a13", 10, '2') -> 'a13'
+pub fn replace_nth_char(s: &str, idx: usize, newchar: char) -> String {
+    s.chars()
+        .enumerate()
+        .map(|(i, c)| if i == idx { newchar } else { c })
+        .collect()
+}
+
+/// Return prefix of string.
+/// "a" -> "b"
+/// "1" -> "2"
+/// [96,97,127] -> [96,98,127]
+/// [127] -> [127, 127]
+/// [127,127,127, 127] -> [127,127,127, 127, 127]
+pub fn prefix_of_string(s: &str) -> String {
+    let mut l = s.len();
+    while l > 0 {
+        l -= 1;
+        if let Some(c) = s.chars().nth(l) {
+            if c == 127 as char {
+                continue;
+            }
+            return replace_nth_char(s, l, (c as u8 + 1) as char);
+        }
+    }
+    format!("{}{}", s, 127 as char)
+}
```

Comment:
Why use `127` as the greatest char? Is it a concern about `utf` string?

Reply:
> Why use `127` as the greatest char? Is it a concern about `utf` string?

I think support ASCII character is enough...

Reply:
> > Why use `127` as the greatest char? Is it a concern about `utf` string?
> 
> I think support ASCII character is enough...

Hmm... I agree to introduce a limit of only allowing to use ascii string.
With this assumption, it should return some error if the input string is not ascii string.

BTW, `String::chars()` returns a iterator of unicode characters:

```rust
    let a = "我".to_string();
    let x = a.chars();
    for c in x {
        println!("{:?}, {:?}", c, char::MAX);
        // '我', '\u{10ffff}'
    }

```